tools/render/main.cpp

```
@@ -1,22 +1,111 @@
 }  // namespace
 
 int main(int argc, char** argv) {
   if (argc != kExpectedArgCount) {
-    std::fprintf(stderr, "usage: render \"<share code>\" out/<name>.wav\n");
+    std::fprintf(stderr, "usage: render \"<share code>\" <cycles> out/<name>.wav\n");
+    return kExitUsage;
+  }
+  char* end = nullptr;
+  const long cycles = std::strtol(argv[2], &end, 10);
+  if (end == argv[2] || *end != '\0' || cycles < 1 || cycles > render::kMaxCycles) {
+    std::fprintf(stderr, "render: cycles must be 1 to %d, got %s\n", render::kMaxCycles, argv[2]);
```

Clang (14.0.6)

[warning] do not call c-style vararg functions

(cppcoreguidelines-pro-type-vararg)